src/platform_impl/orbital/window.rs

```diff
@@ -359,30 +414,58 @@ impl Window {
     }
 
     #[inline]
-    pub fn set_cursor_grab(&self, _: window::CursorGrabMode) -> Result<(), error::ExternalError> {
-        Err(error::ExternalError::NotSupported(
-            error::NotSupportedError::new(),
-        ))
+    pub fn set_cursor_grab(
+        &self,
+        mode: window::CursorGrabMode,
+    ) -> Result<(), error::ExternalError> {
+        let (grab, relative) = match mode {
+            window::CursorGrabMode::None => (false, false),
+            window::CursorGrabMode::Confined => (true, false),
+            window::CursorGrabMode::Locked => (true, true),
+        };
+        self.window_socket
+            .write(format!("M,G,{}", if grab { 1 } else { 0 }).as_bytes())
+            .map_err(|err| error::ExternalError::Os(os_error!(OsError::new(err))))?;
+        self.window_socket
+            .write(format!("M,R,{}", if relative { 1 } else { 0 }).as_bytes())
+            .map_err(|err| error::ExternalError::Os(os_error!(OsError::new(err))))?;
```

Comment:
What relative means here? Locked is the cursor that stays at the same place and not moved. Confined is the one that clamps to window boundaries. The same as in pointer contraints protocol on wayland.

Reply:
Relative means that the cursor will not move and relative mouse events will be sent instead of absolute ones.

Reply:
I am converting back to a draft because the moue relative event is not handled yet, so I want to handle that properly.

Reply:
aye, but you don't need _both_ working at the same time, you can just tell that something is not supported.

Reply:
Latest commit emits `DeviceEvent::MouseMotion` for relative mouse events, so this will work properly.